File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        n = len(nums)
        mp ={}
        preSum =0
        count =0

        mp[0]=1

        for i in range(n):
            preSum += nums[i]
            remove = preSum -k
            if remove in mp:
                count += mp[remove]
            if preSum in mp:
                mp[preSum]+=1
            else:
                mp[preSum]=1
        return count

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int,int> mp;
        int preSum =0;
        int count = 0;

        mp[0]=1;

        for(int i=0;i<n;i++){
            preSum += nums[i];

            if(mp.find(preSum-k)!=mp.end()){
                count += mp[preSum-k];
            }

            mp[preSum]++;
        }
        return count;
    }
};